tests/session/session-pin-001.phpt

```diff
@@ -0,0 +1,56 @@
+--TEST--
+MongoDB\Driver\Session test: executeBulkWrite pins transaction to server
+--SKIPIF--
+<?php require __DIR__ . "/../utils/basic-skipif.inc"; ?>
+<?php skip_if_not_mongos_with_replica_set(); ?>
+<?php skip_if_server_version('<', '4.1.6'); ?>
+<?php skip_if_not_clean(); ?>
+--FILE--
+<?php
+require_once __DIR__ . "/../utils/basic.inc";
+
+$manager = new MongoDB\Driver\Manager(URI);
+
+/* Create collections as that can't be (automatically) done in a transaction */
+$manager->executeCommand(
+    DATABASE_NAME,
+    new \MongoDB\Driver\Command([ 'create' => COLLECTION_NAME ]),
+    [ 'writeConcern' => new \MongoDB\Driver\WriteConcern( \MongoDB\Driver\WriteConcern::MAJORITY ) ]
+);
+
+$session = $manager->startSession();
+var_dump($session->getServer() instanceof \MongoDB\Driver\Server);
+
+$session->startTransaction();
+var_dump($session->getServer() instanceof \MongoDB\Driver\Server);
+
+$bulk = new MongoDB\Driver\BulkWrite();
+$bulk->insert(['x' => 1]);
+$manager->executeBulkWrite(NS, $bulk, ['session' => $session]);
+
+var_dump($session->getServer() instanceof \MongoDB\Driver\Server);
+
+$bulk = new MongoDB\Driver\BulkWrite();
+$bulk->insert(['x' => 1]);
+$manager->executeBulkWrite(NS, $bulk, ['session' => $session]);
```

Comment:
May be worth also asserting that the pinned server remains the same. You can rely on the [Server's comparison handler](https://github.com/mongodb/mongo-php-driver/blob/1.6.0alpha3/src/MongoDB/Server.c#L523) (using `==` of course), which does a string comparison of the host and port. Likewise for other pinning tests below.

I don't quite recall why we don't simply compare numeric server IDs there, but you can open a ticket about that if you're also curious and would like to investigate at some point.

Reply:
Comparing that the correct server is pinned is a good idea, I added that.

I've created [PHPC-1429](https://jira.mongodb.org/browse/PHPC-1429) to track optimising the logic of the comparison handler.